File: 04 Kontrollstrukturen/Pascal/Aufgabe5.pas
Program Aufgabe5;

{$codepage utf8}

Var Eingabe: String;
    Tage: Integer;
    Fehler: Boolean = False;

Begin
    Write('Bitte gib den Monat ein ("jan", "feb", "mrz", ...): ');
    Read(Eingabe);

    Case Eingabe Of
        'jan', 'mrz', 'mai', 'jul', 'aug', 'okt', 'dez': Tage := 31;
        'apr', 'jun', 'sep', 'nov': Tage := 30;
        'feb': Tage := 28;
    Else
        Fehler := True;
    End;

    If Fehler Then
        WriteLn('Es gab einen Fehler bei der Eingabe: ', Eingabe, ' ist kein Monat!')
    Else
        WriteLn('Der Monat ', Eingabe, ' hat ', Tage, ' Tage.');
End.

File: 04 Kontrollstrukturen/Go/Aufgabe5.go
package main

import "fmt"

var (
	eingabe string
	tage int
	fehler bool = false
)

func main() {
	fmt.Print("Bitte gib den Monat ein (\"jan\", \"feb\", \"mrz\", ...): ")
	fmt.Scan(&eingabe)

	switch eingabe {
		case "jan", "mrz", "mai", "jul", "aug", "okt", "dez":
			tage = 31
		case "apr", "jun", "sep", "nov":
			tage = 30
		case "feb":
			tage = 28
		default:
			fehler = true
	}

	if fehler {
		fmt.Println("Es gab einen Fehler bei der Eingabe:", eingabe, "ist kein Monat!")
	} else {
		fmt.Println("Der Monat", eingabe, "hat", tage, "Tage.");
	}
}